### ISSUE TITLE
Delete assign

### ISSUE THREAD

**Issue by Oupsla** [octokit/octokit.js#431]
Hello 
I noticed an error about the "issues.removeAssigneesFromIssue".
I think that the api send the parameter as query and not in the body as json so Github return a 'Error parsing JSON'.

With node-github (not working) :

```
github.issues.removeAssigneesFromIssue({
        user: "user",
        repo: "repo",
        number: "1",
        assignees: "['userAssign']"
      }, function(err, res) {
          done(err, res);
      });
```

With require (working) :

```

var body = '{"assignees": ["'+userAssign+'"]}';

    var options = {
      url: "https://api.github.com/repos/user/repo/issues/1/assignees?access_token=" + token,
      method: "DEL",
      headers: {
        'User-Agent': 'MyApp-App'
      },
      json: JSON.parse(body)
    };

    request.del(options,function(error,response,body){
      if(error){
        console.log(error);
      }else{
        console.log(response);
      }
    });
```

Github link : https://developer.github.com/v3/issues/assignees/#remove-assignees-from-an-issue

**Comment by kaizensoze** [comment]
Should be fixed in v4.1.0.

**Comment by bkeepers** [comment]
Event with v4.1.0, I can't find an incantation that makes this work. I tried calling `github.issues.removeAssigneesFromIssue` with all of these variations:

``` js
{body:JSON.stringify({assignees:['bkeepers']})}
{body:JSON.stringify(['bkeepers'])}
{body:{assignees:['bkeepers']}}
{body:['bkeepers']}
```

All of those result in:

```
{"message":"Problems parsing JSON","documentation_url":"https://developer.github.com/v3"}
```

**Comment by kaizensoze** [comment]
Whoops, there's a bug in the codebase that assumes head|get|delete requests won't have a request body. Will have a fix out shortly.

**Comment by kaizensoze** [comment]
Ok should be fixed in 5.0.1. Added an [example](https://github.com/mikedeboer/node-github/blob/master/examples/removeAssigneesFromIssue.js).

**Comment by brunolemos** [comment]
This also happens on PUT requests like `github.activity.markNotificationsAsReadForRepo({ owner: 'facebook', repo: 'react' })`
